src/common/api/main/WorkerClient.ts

```diff
@@ -40,7 +41,7 @@ export class WorkerClient {
 		return this._deferredInitialized.promise
 	}
 
-	async init(locator: IMainLocator): Promise<void> {
+	async init(locator: IMailLocator | ICalendarLocator): Promise<void> {
```

Comment:
ideally the code in common should not know about any locator but the common one. Was it necessary here?

Reply:
Nope, good call